### PR TITLE
SQL Server integration

### LINKED ISSUE #59: SQL Server integration
Hi,

I've added SQL Server as an option in a fork.  Before I submit the PR, I have a question regarding options.  It seems the status quo is to provide relatively few options.  I think for SQL Server, adding `instance` would be the bare minimum, but my instinct was to add a few more.  Which way do you prefer to lean, i.e. would you want `ssl` and/or the bottom row in or out?

image

### PR DESCRIPTION
**Pull request by simonmcconnell**
adds SQL Server as an option for Connection and SQL cells.  Closes #59

### CODE REVIEW

File: lib/kino_db/connection_cell.ex [hunk 1]
```diff
@@ -137,6 +138,11 @@ defmodule KinoDB.ConnectionCell do
             do: ~w|database schema account username password_secret|,
             else: ~w|database schema account username password|
 
+        "sqlserver" ->
+          if fields["use_password_secret"],
+            do: ~w|database hostname port use_ipv6 username password_secret ssl instance|,
+            else: ~w|database hostname port use_ipv6 username password ssl instance|
```
Comment (jonatanklosko) [inline lib/kino_db/connection_cell.ex:144]:
```suggestion
            do: ~w|database hostname port use_ipv6 username password_secret use_ssl instance|,
            else: ~w|database hostname port use_ipv6 username password use_ssl instance|
```

File: lib/kino_db/sql_cell.ex [hunk 2]
```diff
@@ -292,45 +297,54 @@ defmodule KinoDB.SQLCell do
...
   end
 
-  defp parameterize(<<char::utf8, rest::binary>>, raw, params, n, next) do
-    parameterize(rest, <<raw::binary, char::utf8>>, params, n, next)
+  defp parameterize(<<char::utf8, rest::binary>>, raw, params, n, type, next) do
+    parameterize(rest, <<raw::binary, char::utf8>>, params, n, type, next)
+  end
+
+  defp quote_param("sqlserver", inner, sql_param) do
+    Code.string_to_quoted("%Tds.Parameter{name: \"#{sql_param}\", value: #{inner}}")
```
[the remark this answers is not in the document]
Comment (simonmcconnell) [reply lib/kino_db/sql_cell.ex:343]: I wasn't sure which way to do this bit. Don't we need to wrap it in `Code.ensure_loaded?` or similar so we know `Tds` is available if we do it this way? Most people won't have it as a dep.

[the remark this answers is not in the document]
Comment (jonatanklosko) [reply lib/kino_db/sql_cell.ex:343]: Code inside `quote` is converted to the AST at compilation time and the regular compilation does not apply:

```
iex(1)> %Tds.Parameter{}
error: Tds.Parameter.__struct__/1 is undefined, cannot expand struct Tds.Parameter. Make sure the struct name is correct. If the struct name exists and is correct but it still cannot be found, you likely have cyclic module usage in your code
  iex:1

** (CompileError) cannot compile code (errors have been logged)

iex(1)> quote do: %Tds.Parameter{}
{:%, [], [{:__aliases__, [alias: false], [:Tds, :Parameter]}, {:%{}, [], []}]}
```

**Review by jonatanklosko — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**